phpdotnet/phd/Package/Generic/XHTML.php

```diff
@@ -1846,6 +1848,32 @@ public function format_orderedlist($open, $name, $attrs, $props) {
         return '</ol>';
     }
 
+    /* Support for key inputs is coded like junk */
+    public function format_keycap($open, $name, $attrs, $props) {
+        if ($open) {
+            $content = '';
+            if ($props['sibling']) {
+                $content = ' + ';
+            }
+            return $content . '<kbd class="' . $name . '">';
+        }
+        return '</kbd>';
+    }
+
+    public function format_keycombo($open, $name, $attrs, $props) {
+
+        if (isset($attrs[Reader::XMLNS_DOCBOOK]["action"])) {
+            if ($attrs[Reader::XMLNS_DOCBOOK]["action"] !== "simul") {
+                v('No support for keycombo action = ' . $attrs[Reader::XMLNS_DOCBOOK]["action"],E_USER_WARNING);
```

Comment:
```suggestion
                v('No support for keycombo action = %s', $attrs[Reader::XMLNS_DOCBOOK]["action"], E_USER_WARNING);
```

Nits: prefer parameterised message, and space after comma.